aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,609 @@
+        } = move_from<DAO>(dao);
+       dao_signer_capability
+    }
+
+
+    /// Get the proposal
+    public fun get_proposal(proposal_id: u64, nft_dao: address): Proposal acquires Proposals {
+        assert!(exists<Proposals>(nft_dao), error::not_found(EPRPOSALS_NOT_EXIST_AT_ADDRESS));
+        let proposals = &borrow_global<Proposals>(nft_dao).proposals;
+        assert!(table::contains(proposals, proposal_id), error::not_found(EPRPOSAL_ID_NOT_EXIST));
+        *table::borrow(proposals, proposal_id)
+    }
+
+    /// Get the proposal resolution result
+    public fun get_proposal_resolution(proposal_id: u64, nft_dao: address): u8 acquires Proposals {
+        let proposal = get_proposal(proposal_id, nft_dao);
+        proposal.resolution
+    }
+
+    /// Unpack the proposal fields
+    public fun unpack_proposal(proposal: &Proposal): (String, String, String, PropertyMap, u64, u8) {
+        (
+            proposal.name,
+            proposal.description,
+            proposal.function_name,
+            proposal.function_args,
+            proposal.start_time_sec,
+            proposal.resolution,
+        )
+    }
+
+    /// Unpack the DAO fields
+    public fun unpack_dao(nft_dao: address): (String, u64, address, String, u64, u64, u64) acquires DAO {
+        let dao = borrow_global_mut<DAO>(nft_dao);
+        (
+            dao.name,
+            dao.resolve_threshold,
+            dao.governance_token.creator,
+            dao.governance_token.collection,
+            dao.voting_duration,
+            dao.min_required_proposer_voting_power,
+            dao.next_proposal_id,
+        )
+    }
+
+    /// Get owned DAOs of an admin
+    public fun get_owned_daos(account: &signer): vector<address> acquires OwnedDAO {
+        let addr = signer::address_of(account);
+        assert!(exists<OwnedDAO>(addr), error::not_found(EDAO_NOT_EXIST));
+        let owned_dao = borrow_global<OwnedDAO>(addr);
+        owned_dao.daos
+    }
+
+    // transfer APT fund from the DAO account to the destination account
+    fun transfer_fund(res_acct: &signer, dst: address, amount: u64) {
+        coin::transfer<AptosCoin>(res_acct, dst, amount);
+    }
+
+    fun get_voting_weights(
+        account: &signer,
+        token_names: &vector<String>,
+        property_versions: &vector<u64>,
+        dao: &DAO
+    ): u64 {
+        let gtoken = &dao.governance_token;
+        let i = 0;
+        let total_weight = 0;
+        while (i < vector::length(token_names)) {
+            let token_name = *vector::borrow(token_names, i);
+            let property_version = *vector::borrow(property_versions, i);
+            let token_id = token::create_token_id_raw(gtoken.creator, gtoken.collection, token_name, property_version);
+            assert!(token::balance_of(signer::address_of(account), token_id) == 1, error::permission_denied(ENOT_OWN_THE_VOTING_DAO_TOKEN));
+            total_weight = total_weight + if (is_global_unique_token(token_id)) {1} else {0};
```

Comment:
Why not just

if (is_global_unique_token(token_id)) {
  total_weight = total_weight + 1;
}

Reply:
personal preference of reading inline. But updated.